### PR TITLE
Enhanced RDoc for IO

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Treats:
- #lineno
- #lineno=
- #readline
- #readlines
- #each_line
- #each_byte
- #each_char
- #each_codepoint

### CODE REVIEW

File: io.c [hunk 1]
```diff
@@ -4127,26 +4107,57 @@ static VALUE io_readlines(const struct getline_arg *arg, VALUE io);
...
  *
- *     f = File.new("testfile", chomp: true)
- *     f.readlines[0]   #=> "This is line one"
+ *    f = File.new('t.txt')
+ *    f.readlines
+ *    # => ["First line\n", "Second line\n", "\n", "Fourth line\n", "Fifth line\n"]
+ *    f.readlines # => []
+ *
+ *  With only string argument +sep+ given,
+ *  returns the next line as determined by line separator +sep+,
```
Comment (jeremyevans) [inline io.c:4127]: `returns the next line` seems wrong, since it returns an array of lines.

File: io.c [hunk 2]
```diff
@@ -4127,26 +4107,57 @@ static VALUE io_readlines(const struct getline_arg *arg, VALUE io);
...
+ *    # => ["First line\nSecond line\n\n", "Fourth line\nFifth line\n"]
+ *
+ *  With only integer argument +limit+ given,
+ *  limits the number of bytes in each line;
+ *  see {Line Limit}}[#class-IO-label-Line+Limit]:
+ *
+ *    f = File.new('t.txt')
+ *    f.readlines(8)
+ *    # => ["First li", "ne\n", "Second l", "ine\n", "\n", "Fourth l", "ine\n", "Fifth li", "ne\n"]
+ *
```
Comment (jeremyevans) [inline io.c:4153]: Probably should add a discussion of when `sep` and `limit` are both provided.

File: io.c [hunk 3]
```diff
@@ -4174,33 +4185,101 @@ io_readlines(const struct getline_arg *arg, VALUE io)
...
+ *    "First li"
+ *    "ne\n"
+ *    "Second l"
+ *    "ine\n"
+ *    "\n"
+ *    "Fourth l"
+ *    "ine\n"
+ *    "Fifth li"
+ *    "ne\n"
+ *
```
Comment (jeremyevans) [inline io.c:4264]: Likewise, a section about both `sep` and `limit` would be good to add here.

File: io.c [hunk 4]
```diff
@@ -13864,12 +13952,12 @@ set_LAST_READ_LINE(VALUE val, ID _x, VALUE *_y)
  *  Also available are the options offered in String#encode,
  *  which may control conversion between external internal encoding.
  *
- *  == Getline Options
+ *  == Line Options
  *
  *  A number of \IO methods accept optional keyword arguments
- *  that determine how a stream is to be treated:
+ *  that determine how lines in a stream is to be treated:
```
Comment (jeremyevans) [inline io.c:13958]: `is to be` -> `are to be`

**Review by jeremyevans — CHANGES_REQUESTED** [review]
Looks pretty good, just a few changes requested.

File: io.c (hunk 1 above)
Comment (BurdetteLamar) [reply io.c]: Done.

File: io.c (hunk 2 above)
Comment (BurdetteLamar) [reply io.c:4160]: Done.

File: io.c (hunk 4 above)
Comment (BurdetteLamar) [reply io.c]: Done.

File: io.c (hunk 3 above)
Comment (BurdetteLamar) [reply io.c:4277]: Done, plus a couple of others.

**Review by jeremyevans — APPROVED** [review]
Looks good!

**Pull request merged**